backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/handlingunits/HUEditorRowAttributes.java

```diff
@@ -121,8 +121,9 @@ public static HUEditorRowAttributes cast(final IViewRowAttributes attributes)
 		for (final I_M_Attribute attribute : attributes)
 		{
 			final AttributeCode attributeCode = HUEditorRowAttributesHelper.extractAttributeCode(attribute);
+			final boolean alwaysUpdatable = attribute.isAlwaysUpdateable();
 
-			if (readonlyEffective || attributesStorage.isReadonlyUI(calloutCtx, attribute))
+			if ((!alwaysUpdatable && readonlyEffective) || attributesStorage.isReadonlyUI(calloutCtx, attribute))
```

Comment:
* i tried to draw a table, but this expression is too complicated for me please make it clearer
* readonlyEffective seems to be misnamed, same as `isReadonlyUI` (i mean, isReadonlyUI can return true and still it's writable, right?)

Reply:
rewritten, as talked